lib/ansible/modules/monitoring/zabbix/zabbix_user_group.py

```diff
@@ -0,0 +1,331 @@
+
+        changed = self.compare_config(generated_config, current_config)
+
+        if self._module.check_mode:
+            self._module.exit_json(changed=changed)
+
+        if changed:
+            generated_config['usrgrpid'] = group_id
+            self._zapi.usergroup.update(generated_config)
+
+        return changed
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            validate_certs=dict(type='bool', required=False, default=True),
+            name=dict(type='str', required=True),
+            debug_mode=dict(type='bool', default=False, required=False),
+            gui_access=dict(default='default', required=False, choices=['default', 'internal', 'ldap', 'disabled']),
+            users_status=dict(default='enabled', required=False, choices=['enabled', 'disabled']),
+            rights=dict(
+                type='list',
+                default=[],
+                elements='dict',
+                required=False,
+                options=dict(
+                    host_group=dict(type='str', required=True),
+                    permission=dict(type='str', required=True, choices=['denied', 'RO', 'RW'])
+                )
+            ),
+            users=dict(
+                type='list',
+                default=[],
+                required=False,
+                elements='str'
+            ),
+            state=dict(default="present", choices=['present', 'absent']),
+            timeout=dict(type='int', default=10)
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg=missing_required_lib('zabbix-api', url='https://pypi.org/project/zabbix-api/'), exception=ZBX_IMP_ERR)
+
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    validate_certs = module.params['validate_certs']
+    name = module.params['name']
+    debug_mode = module.params['debug_mode']
+    gui_access = module.params['gui_access']
+    users_status = module.params['users_status']
+    users = module.params['users']
+    rights = module.params['rights']
+    state = module.params['state']
+    timeout = module.params['timeout']
+
+    zbx = None
+
+    # Login to zabbix
+    try:
+        zbx = ZabbixAPI(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password, validate_certs=validate_certs)
+        zbx.login(login_user, login_password)
```

Comment:
```suggestion
        zbx.login(login_user, login_password)
        atexit.register(zbx.logout)
```

Please add `atexit.register(zbx.logout)`
see https://github.com/ansible/ansible/pull/58525